libafl/src/feedbacks/map.rs

```diff
@@ -393,11 +371,14 @@ where
         let initial = observer.initial();
 
         let map_state = state
-            .feedback_states_mut()
-            .match_name_mut::<MapFeedbackState<T>>(&self.name)
+            .named_metadata_mut()
+            .get_mut::<MapFeedbackMetadata<T>>(&self.name)
             .unwrap();
+        if map_state.history_map.len() < observer.len() {
```

Comment:
why do we resize here when map_state.history_map.len() < observer.len() ?
I think it should error out just like `assert!(size <= observer.len())`

Reply:
Some map observers can potentially grow over time, the history map has to change accordingly

Reply:
ok now I understand.